examples/model_compress/pruning/pruning_bert_glue.py

```diff
@@ -0,0 +1,355 @@
+            # The final ffn weight remaining ratio is the half of the attention weight remaining ratio.
+            # This is just an empirical configuration, you can use any other method to determine this sparsity.
+            sparsity = 1 - (1 - len(head_idx) / 12) * 0.5
+            # here we use a simple sparsity schedule, we will prune ffn in 12 iterations, each iteration prune `sparsity_per_iter`.
+            sparsity_per_iter = 1 - (1 - sparsity) ** (1 / 12)
+            ffn_config_list.append({'op_names': [f'bert.encoder.layer.{layer_count}.intermediate.dense'], 'sparsity': sparsity_per_iter})
+            layer_count += 1
+
+    attention_pruned_model.bert.encoder.layer = torch.nn.ModuleList(module_list)
+
+    ##########################################################################################################################
+    # Step 3: Retrain the attention pruned model.
+    ##########################################################################################################################
+    total_epochs = 5
+    optimizer = Adam(attention_pruned_model.parameters(), lr=lr, eps=1e-8)
+
+    def lr_lambda(current_step: int):
+        return max(0.0, float(total_epochs * steps_per_epoch - current_step) / float(total_epochs * steps_per_epoch))
+
+    lr_scheduler = LambdaLR(optimizer, lr_lambda)
+    at_model_save_path = log_dir / 'attention_pruned_model_state.pth'
+    training(attention_pruned_model, optimizer, fake_criterion, lr_scheduler=lr_scheduler, max_epochs=total_epochs, save_best_model=True, save_path=at_model_save_path, teacher_model=teacher_model)
+
+    attention_pruned_model.load_state_dict(torch.load(at_model_save_path))
+
+    ##########################################################################################################################
+    # Step 4 & 5: Iterative pruning FFN with TaylorFOWeightPruner in 12 iterations.
+    #             Finetuning 2000 steps after each iteration, then finetuning 2 epochs after pruning finished.
+    ##########################################################################################################################
+
+    distil_training = functools.partial(training, teacher_model=teacher_model, log_path=log_dir / 'taylor_pruning.log')
+    traced_optimizer = nni.trace(Adam)(attention_pruned_model.parameters(), lr=3e-5, eps=1e-8)
+    evaluator = TorchEvaluator(distil_training, traced_optimizer, fake_criterion)
+
+    attention_pruned_model.train()
+    current_step = 0
+    best_result = 0
+    total_epochs = 4
+    init_lr = 3e-5
+
+    for current_epoch in range(total_epochs):
+        for batch in train_dataloader:
+            # pruning 12 times
+            if current_step % 2000 == 0 and current_step < 24000:
+                check_point = attention_pruned_model.state_dict()
+                pruner = TaylorFOWeightPruner(attention_pruned_model, ffn_config_list, evaluator, 1000)
```

Comment:
iterative pruner does not work here? so we have to write the loop manually?

Reply:
yes, we need add `step level` to iterative pruner, will do this in v3.0